swagger/src/main/scala/org/http4s/rho/swagger/SwaggerSupport.scala

```diff
@@ -10,38 +9,26 @@ import headers.`Content-Type`
 
 import shapeless.HList
 
-
 trait SwaggerSupport extends RhoService {
+  import models._
+
+  private var swagger: Option[Swagger] = None
 
   /** Override the `swaggerFormats` to add your own custom serializers */
   def swaggerFormats: SwaggerFormats = DefaultSwaggerFormats
 
   def apiPath = "swagger.json"
-  def apiVersion: String = "1.0.0"
-
-  private val swagger = {
-    val swagger = new m.Swagger
-    val info = new m.Info()
-      .version(apiVersion)
-      .title("Swagger Petstore")
-
-    val contact = new m.Contact()
-      .name("Wordnik API Team")
-      .email("foo@bar.baz")
-      .url("http://swagger.io")
-    info.setContact(contact)
-
-    swagger.setInfo(info)
-    swagger
-  }
+
+  def apiInfo: Info = Info(title = "My API", version = "1.0.0")
 
   GET / apiPath |>> { () =>
-    val swaggerJson = Json.mapper().writeValueAsString(swagger)
+    val swaggerJson = Json.mapper().writeValueAsString(swagger.map(_.toJModel).get)
```

Comment:
So, should I apply the changes mentioned above and then make a new pull request again?